File: frontend/src/types/duplicate.ts
import type { Photo } from './photo'

export type DuplicateType = 'exact' | 'similar' | 'burst'

export interface DuplicateGroup {
  id: number
  group_type: DuplicateType
  recommended_keep_id: number | null
  photos: Photo[]
}

export interface ResolveRequest {
  group_id: number
  keep_ids: number[]
  delete_ids: number[]
}

File: backend/app/schemas/duplicate.py
"""
Pydantic schemas for duplicate group responses and resolve requests.
"""
from pydantic import BaseModel, Field

from app.models.duplicate_group import DuplicateType
from app.schemas.photo import PhotoResponse


class DuplicateGroupResponse(BaseModel):
    id: int
    group_type: DuplicateType
    recommended_keep_id: int | None
    photos: list[PhotoResponse]

    model_config = {"from_attributes": True}


class ResolveRequest(BaseModel):
    """
    Request body for POST /api/v1/duplicates/resolve.
    Caller specifies which photo IDs to keep and which to soft-delete.
    """

    group_id: int
    keep_ids: list[int] = Field(description="Photo IDs to mark as kept")
    delete_ids: list[int] = Field(description="Photo IDs to soft-delete")
